internal/gke/deploy.go

```diff
@@ -956,11 +954,90 @@ func createServiceAccountsIAMPolicyRole(ctx context.Context, config CloudConfig)
 	})
 }
 
+// getConfigCluster computes the region and the cluster that manages all the
+// applications within a Service Weaver project, and the cluster .
+func getConfigCluster(ctx context.Context, config CloudConfig, cfg *config.GKEConfig) (string, string, error) {
+	region, name, err := getRunningConfigCluster(config)
+	if err != nil {
+		return "", "", err
+	}
+	if region == "none" {
+		// No config cluster. Set up a config cluster in the first region from the
+		// list of regions where to rollout the app, as specified by the user.
+		return cfg.Regions[0], applicationClusterName, nil
+	}
+
+	// Check if the config cluster actually exists.
+	exists, err := hasCluster(ctx, config, name, region)
+	if err != nil {
+		return "", "", err
+	}
+	if exists {
+		return region, name, nil
+	}
+	// The config cluster doesn't exist; disable ingress.
+	// TODO(rgrandl): we have to force the deletion of the ingress, because some
+	// other resources that depend on the ingress might be lingering around. We
+	// should delete all these resources as well.
+	_, err = runGcloud(config,
+		fmt.Sprintf("Disable multi-cluster ingress for cluster %q in %q",
+			name, region), cmdOptions{},
+		"container", "fleet", "ingress", "disable", "--force")
+
+	// The name of new config clusters will always be applicationClusterName.
+	name = applicationClusterName
+	if err != nil {
+		return "", "", err
+	}
+
+	// Set up the config cluster in the first region from the list of regions
+	// where to rollout the app, as specified by the user.
+	return cfg.Regions[0], name, nil
+}
+
+// getRunningConfigCluster returns information about the config cluster that
+// manages all the applications within a Service Weaver project.
+//
+// Returns none if no config cluster is running.
```

Comment:
Can we have it return `""` if no config cluster is running?

Also, clarify that it returns `"", "", nil`

Finally, can we have it return name, region, error?

Reply:
Done